packages/rs-sdk-trusted-context-provider/src/provider.rs

```diff
@@ -58,19 +59,61 @@ pub struct TrustedHttpContextProvider {
 }
 
 impl TrustedHttpContextProvider {
-    /// Create a new trusted HTTP context provider
+    /// Verify that a URL's domain resolves
+    fn verify_domain_resolves(url: &str) -> Result<(), TrustedContextProviderError> {
+        let parsed_url = Url::parse(url).map_err(|e| {
+            TrustedContextProviderError::NetworkError(format!("Invalid URL: {}", e))
+        })?;
+
+        let host = parsed_url.host_str().ok_or_else(|| {
+            TrustedContextProviderError::NetworkError("URL has no host".to_string())
+        })?;
+
+        let port = parsed_url.port().unwrap_or(443); // Default to HTTPS port
```

Comment:
this should be determined based on Url::scheme() or Url::port_or_known_default()